triage/script.js

```diff
@@ -0,0 +1,738 @@
+var clusteredAll = null;	  // all clusters
+var options = null;			  // user-provided in form or URL
+var lastClusterRendered = 0;  // for infinite scrolling
+
+const kGraphHeight = 200; 	  // keep synchronized with style.css:div.graph
+
+/**********************************************************************
+ *                        POLYFILLS                                   *
+ **********************************************************************/
+
+if (Object.entries === undefined) {
+	// Simple polyfill for Safari compatibility.
+	// Object.entries is an ES2017 feature.
+	Object.entries = function*(obj) {
+		for (let key of Object.keys(obj)) {
+			yield [key, obj[key]];
+		}
+	}
+}
+
+/**********************************************************************
+ *                                 DATASTRUCTURES                     *
+ **********************************************************************/
+
+// Return the minimum and maximum value of arr.
+// Doesn't cause stack overflows like Math.min(...arr).
+function minMaxArray(arr) {
+	var min = arr[0];
+	var max = arr[0];
+	for (var i = 1; i < arr.length; i++) {
+		if (arr[i] < min) min = arr[i];
+		if (arr[i] > max) max = arr[i];
+	}
+	return [min, max];
+}
+
+// Store information about individual builds.
+class Builds {
+	constructor(dict) {
+		this.jobs = dict.jobs;
+		this.jobPaths = dict.job_paths;
+		this.cols = dict.cols;
+		this.colStarted = this.cols.started;
+		this.colPr = this.cols.pr;
+		this.timespan = minMaxArray(this.cols.started);
+		this.runCount = this.cols.started.length;
+	}
+
+	// Create a build object given a job and build number.
+	get(job, number) {
+		let indices = this.jobs[job];
+		if (indices.constructor === Array) {
+			let [start, count, base] = indices;
+			if (number < start || number > start + count) {
+				console.error('job ' + job + ' number ' + number + ' out of range.');
+				return;
+			}
+			var index = base + (number - start);
+		} else {
+			var index = indices[number];
+		}
+		// Add more columns as necessary.
+		// This is faster than dynamically adding properties to an object.
+		return {
+			job: job,
+			number: number,
+			started: this.colStarted[index],
+			pr: this.colPr[index],
+		};
+		/* Old way:
+		let build = {
+			job: job,
+			number: number,
+		}
+		for (let [key, value] of Object.entries(this.cols)) {
+			build[key] = value[index];
+		}
+		return build;
+		*/
+	}
+
+	// Count how many builds a job has.
+	count(job) {
+		let indices = this.jobs[job];
+		if (indices.constructor === Array) {
+			return indices[1];
+		}
+		return Object.keys(indices).length;
+	}
+
+	getStartTime() {
+		return new Date(this.timespan[0] * 1000);
+	}
+
+	getEndTime() {
+		return new Date(this.timespan[1] * 1000);
+	}
+}
+
+function sum(arr, keyFunc) {
+	if (arr.length === 0)
+		return 0;
+	return arr.map(keyFunc).reduce((a, b) => a + b);
+}
+
+var clustersSum = clusters => sum(clusters, c => sum(c[1], t => t[1].length));
```

Comment:
Will you please

a) limit anonymous function depth to 1 (aka do not have an anon function inside an anon function) and
b) consider using `{ }` and indent by two spaces? (like what you're doing with `vals.sort` below

Also any reason not to make this a normal function?